boards/p1am_100/src/lib.rs

```diff
@@ -0,0 +1,297 @@
+        }
+    }
+    PA15 {
+        name: pa15,
+        aliases: {
+            AlternateC: SdSdi
+        }
+    }
+    PA16 {
+        name: d8,
+        aliases: {
+            AlternateC: Spi0Sdo
+        }
+    }
+    PA17 {
+        name: d9,
+        aliases: {
+            AlternateC: Spi0Sck
+        }
+    }
+    PA18 {
+        name: pa18,
+        aliases: {
+            /// Host Enable, drive high to switch into USB host mode
+            #[cfg(feature = "usb")]
+            PushPullOutput: UsbId
+        }
+    }
+    PA19 {
+        name: d10,
+        aliases: {
+            AlternateC: Spi0Sdi
+        }
+    }
+    PA20 { name: d6 }
+    PA21 { name: d7 }
+    PA22 { name: d0 }
+    PA23 { name: d1 }
+    PA24 {
+        name: usb_dm,
+        aliases: {
+            #[cfg(feature = "usb")]
+            AlternateG: UsbDm
+        }
+    }
+    PA25 {
+        name: usb_dp,
+        aliases: {
+            #[cfg(feature = "usb")]
+            AlternateG: UsbDp
+        }
+    }
+    PA27 {
+        name: pa27,
+        aliases: {
+            PullUpInput: SdCardDetect
+        }
+    }
+    PA28 {
+        name: switch,
+        aliases: {
+            PullUpInput: Switch
+        }
+    }
+    PB02 {
+        name: d16,
+        aliases: {
+            AlternateB: A1
+        }
+    }
+    PB03 {
+        name: d17,
+        aliases: {
+            AlternateB: A2
+        }
+    }
+    PB08 {
+        name: led,
+        aliases: {
+            PushPullOutput: Led
+        }
+    }
+    PB09 {
+        name: pb09,
+        aliases: {
+            PushPullOutput: BaseEnable,
+            AlternateB: AdcBattery
+        }
+    }
+    PB10 { name: d4 }
+    PB11 { name: d5 }
+    PB22 {
+        name: d14,
+        aliases: {
+            AlternateD: UartTx
+        }
+    }
+    PB23 {
+        name: d13,
+        aliases: {
+            AlternateD: UartRx
+        }
+    }
+);
+
+/// Convenience for setting up the labelled SPI peripheral.
+/// This powers up SERCOM4 and configures it for use as an
+/// SPI Master in SPI Mode 0.
+pub fn spi_master<F: Into<Hertz>>(
+    clocks: &mut GenericClockController,
+    bus_speed: F,
+    sercom4: pac::SERCOM4,
+    pm: &mut pac::PM,
+    sck: gpio::Pb11<Input<Floating>>,
+    mosi: gpio::Pb10<Input<Floating>>,
+    miso: gpio::Pa12<Input<Floating>>,
+    port: &mut Port,
+) -> SPIMaster4<
+    hal::sercom::Sercom4Pad0<gpio::Pa12<gpio::PfD>>,
+    hal::sercom::Sercom4Pad2<gpio::Pb10<gpio::PfD>>,
+    hal::sercom::Sercom4Pad3<gpio::Pb11<gpio::PfD>>,
+> {
+    let gclk0 = clocks.gclk0();
+    SPIMaster4::new(
+        &clocks.sercom4_core(&gclk0).unwrap(),
+        bus_speed.into(),
+        hal::hal::spi::Mode {
+            phase: hal::hal::spi::Phase::CaptureOnFirstTransition,
+            polarity: hal::hal::spi::Polarity::IdleLow,
+        },
+        sercom4,
+        pm,
+        (miso.into_pad(port), mosi.into_pad(port), sck.into_pad(port)),
+    )
+}
```

Comment:
You may want to consider using the new `sercom::v2::spi` driver. Your function could look something like:
``` rust
use sercom::v2::spi;
pub type Spi = spi::Spi<spi::Config<SpiPads, spi::Master, spi::EightBit>>;

pub fn spi_master<F: Into<Hertz>>(
    clock: &mut GenericClockController,
    sercom1: pac::SERCOM4,
    pm: &pac::PM,
    baud: F,
    mosi: Pin<PB10, InputFloating>,
    miso: Pin<PA12, FloatingInput>,
    sclk: Pin<PB11, FloatingInput>,
) -> Spi {
    let core_clock = &clocks.sercom4_core(&clocks.gclk0()).unwrap();
    let pads = sercom::v2::spi::Pads::<Sercom4>::new()
        .sclk::<Pad3, _>(sclk)
        .data_in::<Pad0, _>(miso)
        .data_out::<Pad2, _>(mosi);
    spi::Config::new(pm, sercom1, pads, core_clock.freq())
        .baud(baud)
        .enable()
}
```

Reply:
I recently discovered a way to simplify `spi::Pads`. It's almost ready for a PR, but not quite yet. If you decide to use `v2::spi`, it should only require minor changes.

Reply:
With the latest (unpublished) update to `v2::spi`, you could do this:

```rust
use sercom::v2::spi;
pub type SpiPads = spi::Pads<Sercom4, (Pad0, PB12), (Pad2, PB10), (Pad3, PB11)>;
pub type Spi = spi::Spi<spi::Config<SpiPads>>; // Master & EightBit are the defaults

pub fn spi_master(
    clock: &mut GenericClockController,
    sercom4: pac::SERCOM4,
    pm: &pac::PM,
    baud: impl Into<Hertz>,
    mosi: impl AnyPin<Id = PB10>,
    miso: impl AnyPin<Id = PB12>,
    sclk: impl AnyPin<Id = PB11>,
) -> Spi {
    let core_clock = &clocks.sercom4_core(&clocks.gclk0()).unwrap();
    let pads = spi::Pads::default()
        .sclk(sclk)
        .data_in(miso)
        .data_out(mosi);
    spi::Config::new(pm, sercom4, pads, core_clock.freq())
        .baud(baud)
        .enable()
}
```

Reply:
I think I've cleaned this up significantly, and using the proper pin aliases to boot (instead of `AnyPin`). I was able to avoid duplicating any information in the convenience functions that was already present in the `bsp_pins!` invocation.